src/ig-template/IgtGame.ts

```diff
@@ -27,6 +28,8 @@ export abstract class IgtGame {
    */
   protected abstract readonly TICK_DURATION: number;
 
+  protected _onTick = new EventDispatcher<IgtGame, number>();
```

Comment:
Why do you want to send the entire game with the event? If consumers need access to it they can store it themselves

Reply:
When I initially did this I was unfamiliar with events and how they work, so I went with the 'standard' method of creating, sending, and receiving an event, which involves sending the sender along with the event information, hence the whole game along with the event. I've gone ahead and switched that over to a simple event. I can additionally confirm that for what I need the event for in the first place, you are right that sending the whole game is entirely unnecessary.